storagenode/contact/chore.go

```diff
@@ -48,53 +58,108 @@ func (chore *Chore) Run(ctx context.Context) (err error) {
 	defer mon.Task()(&ctx)(&err)
 	chore.log.Info("Storagenode contact chore starting up")
 
+	var group errgroup.Group
+
 	if !chore.service.initialized.Wait(ctx) {
 		return ctx.Err()
 	}
+	chore.mu.Lock()
+	for _, satellite := range chore.trust.GetSatellites(ctx) {
+		satellite := satellite
+		backOff := time.Second
+		interval := chore.interval
+		cycle := sync2.NewCycle(interval)
+		chore.cycles = append(chore.cycles, cycle)
+
+		cycle.Start(ctx, &group, func(ctx context.Context) error {
```

Comment:
I think we might be able to simplify this block a bit by putting everything into the for loop
```
for _, satellite := range chore.trust.GetSatellites(ctx) {
		satellite := satellite
		backOff := time.Second
		interval := chore.interval
		cycle := sync2.NewCycle(interval)
		chore.cycles = append(chore.cycles, cycle)

		cycle.Start(ctx, &group, func(ctx context.Context) error {
			chore.log.Info("starting contact cycle for satellite " + satellite.String())
			interval := backOff
                         attempts := 0
			for {
                                 err := chore.pingSatellite(ctx, satellite)
				attempts++
                                 if err == nil {
					return nil
				}
                                 chore.log.Error("ping satellite failed " + strconv.Itoa(attempts) + " times")
				interval *= 2
				if interval > chore.interval {
					interval = chore.interval
				}
				if !sync2.Sleep(ctx, interval) {
					chore.log.Error("ping satellite failed after " + strconv.Itoa(attempts) + " attempts timed out")
					return ctx.Err()
				}
			}
		})
	}
```